presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -806,6 +802,26 @@ public void finishInsert(ConnectorSession session, ConnectorInsertTableHandle in
             for (CompletableFuture<?> fileRenameFuture : fileRenameFutures) {
                 MoreFutures.getFutureValue(fileRenameFuture, PrestoException.class);
             }
+
+            // clean remaining partition directories
+            Optional<Path> writePathRoot = locationService.writePath(handle.getLocationHandle(), Optional.empty());
+            if (writePathRoot.isPresent() && !writePathRoot.get().equals(locationService.targetPathRoot(handle.getLocationHandle()))) {
+                for (PartitionUpdate partitionUpdate : partitionUpdates) {
+                    if (writePathRoot.isPresent() && !partitionUpdate.getTargetPath().equals(partitionUpdate.getWritePath())) {
+                        String writePath = partitionUpdate.getWritePath();
+                        checkState(writePath.startsWith(writePath), "Partition updated was write stored outside insert write path");
+                        while (!writePath.equals(writePathRoot.get().toString())) {
+                            if (!deleteIfExists(session.getUser(), writePath)) {
+                                break;
+                            }
+                            writePath = Paths.get(writePath).getParent().toString();
+                        }
+                    }
+                }
+                if (!deleteIfExists(session.getUser(), writePathRoot.get().toString())) {
+                    log.debug("Unable to delete insert temporary directory: '%s'", writePathRoot);
```

Comment:
`Unable to delete temporary directory for insert`. The original sentence is harder to read because of `delete insert`.